### ISSUE TITLE
すぐに動く docker image(ないなら,docker-compose.yml) がほしい。 nginx がいいです。 

### ISSUE THREAD

**Issue by zanjibar** [opensource-workshop/connect-cms-ideas#46]

**Comment by gakigaki** [comment]
@zanjibar 
Docker Hubでイメージを共有する感じですか？

今、個人的にはDocker使って開発環境を動かしています。
ただし、apacheです。

**Comment by gakigaki** [comment]
Docker Hubにあげるとすると、
githubのマスターにプッシュされたら、自動でDocker Hubを更新する仕組みが必要そうですね。

**Comment by zanjibar** [comment]
docker-compose の中に git clone いれておけばいいかなと思いますが、難しいのでしょうか？

**Comment by gakigaki** [comment]
やりたいことを整理させてください。
Connect-CMSのコンテナイメージがほしいわけでなく、
Connect-CMSを動かせるDocker Composeの設定ファイルが欲しいってことですか？

docker imageが欲しいということで、wordpressのコンテナイメージのようなものだと私は認識していました。
https://hub.docker.com/_/wordpress

**Comment by zanjibar** [comment]
はい、image もあるといいですが、その前のステップとして、docker-composer.yml があればよいかなと思っています。それがあれば、image はすぐにつくれるので、

**Comment by zanjibar** [comment]
とりあえず、apache でなんとかログインページまででるところまでいきました。

**Comment by nagasheep** [comment]
@zanjibar 
おはようございます。
永原です。

次回（7/21）のConnect-CMS勉強会で、この件、発表（紹介）していただきたいのですけど、お願いできます？
永原は、Docker の便利さなんかもわかっていないので、その辺も教えていただけると、とてもうれしいです！

**Comment by zanjibar** [comment]
7/21 なら結構余裕かなと思います。

docker ですが、
使ったことないとわからないと思います。
テストというか、CI のときに使うのがよくあるかと思いいます。
github に push するとすぐに、docker でテストできるできるとような感じです。
connect-cms のビジネス展開も、docker 納品で、修正は、github でpush するだけというのが十分にありかなと思っています。
そのときに、クリテリカルなモジュールは、汎用データベースです。

学校向けのリプレースで稼ぎつつ、イントラのデータベースの窓という形の２面作戦です。

企業内のシステムには一切触りたくないけど、データベースの項目を増やして、業務をしたいというニーズはあるはずです。
追加の項目は、connect-cms 側にもたせます。

**Comment by nagasheep** [comment]
@zanjibar さんと調整して、次回に発表していただけるということです！
ありがとうございます。

以下に予定を書きました。
https://connect-cms.jp/plugin/databases/publicConfirm/96/216/93#frame-216

**Comment by zanjibar** [comment]
@gakigaki
docker-compose.yml 作成中です。dbサービスをわけているので、dbがあがってくるのをまって、laravel のインストールをすすめる必要あります。どうやって回避するのがいいでしょうか？　
いくつか解決策があります。
 - 他のサービスの稼働を素朴に待つ
https://kotaroooo0-dev.hatenablog.com/entry/2020/07/25/000000 
- db を build 時に、インストールする
- 手動で動かして、イメージを export して使う
- その他

**Comment by gakigaki** [comment]
@zanjibar 
```
php artisan migrate
php artisan db:seed
```
これらのコマンドに関わる懸念ですね。
いまは手動で動かして回避しています。

**Comment by zanjibar** [comment]
> @zanjibar
> 
手作業なんですね。ちょっと待つというので対応してしてみます。

> php artisan migrate
> php artisan db:seed
> ```
> 
> これらのコマンドに関わる懸念ですね。
> いまは手動で動かして回避しています。

**Comment by zanjibar** [comment]
https://connect-cms.slack.com/archives/C017172QSFQ/p1626852651000100 に資料アップしました。 @gakigaki @akagane99 @akagane99
